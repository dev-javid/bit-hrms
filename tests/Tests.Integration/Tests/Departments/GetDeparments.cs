namespace Tests.Integration.Tests.Departments
{
    public class GetDeparments(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/departments?page=1&limit=10";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Data()
        {
            await LoginAsCompanyAdminAsync();
            await FeedDataAsync("Tests/Departments/GetDeparments.sql");
            using HttpResponseMessage response = await Client.GetAsync(Route);

            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result);
        }
    }
}
