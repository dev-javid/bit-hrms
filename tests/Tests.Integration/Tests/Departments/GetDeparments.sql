INSERT INTO public.departments (id, company_id, name, created_by, created_on)
VALUES
(1, 999, 'HR',          1, now()),
(2, 999, 'Finance',     1, now()),
(3, 999, 'Engineering', 1, now()),
(4, 998, 'Marketing',   1, now()),
(5, 998, 'Sales',       1, now()),
(6, 998, 'Support',     1, now()),
(7, 998, 'IT',          1, now()),
(8, 998, 'Legal',       1, now()),
(9, 998, 'Operations',  1, now());