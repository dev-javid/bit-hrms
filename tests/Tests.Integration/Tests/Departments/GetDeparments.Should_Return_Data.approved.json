{
    "page": 1,
    "limit": 10,
    "total": 4,
    "items": [
        {
            "departmentId": 1,
            "name": "HR"
        },
        {
            "departmentId": 2,
            "name": "Finance"
        },
        {
            "departmentId": 3,
            "name": "Engineering"
        },
        {
            "departmentId": 3000,
            "name": "Test-Department"
        }
    ]
}